docs/features/highlight.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Highlight
+category: features
+---
+
+{@snippet features/build-highlight-source}
+
+The {@link module:highlight/highlight~Highlight} feature offers a text marking tools that help content authors speed up their work, e.g. reviewing content or marking it for the future reference. It uses inline `<marker>` elements in the view, supports both markers (background color) and pens (text color), and comes with a flexible configuration.
+
+## Demo
+
+{@snippet features/highlight}
+
+## Configuring the highlight options
+
+It is possible to configure which highlight options are supported by the editor.
+You can use the {@link module:highlight/highlight~HighlightConfig#options `highlight.options`} configuration and define your own highlight styles.
+
+For example, the following editor supports only two styles (a green marker and a blue pen):
+
+```js
+ClassicEditor
+	.create( document.querySelector( '#editor' ), {
+		highlight: {
```

Comment:
Since the highlight feature dropdown uses a basic `ToolbarView`, I wonder if we could simply make the highlight configuration compatible with the toolbar config. I mean here a support for toolbar separators (`|`), that developers may want to use to group the buttons if only to separate pens from markers. WDYT?

![image](https://user-images.githubusercontent.com/1099479/36369051-227d5552-155a-11e8-94f3-2282a7ec0ba7.png)

Reply:
Very interesting idea. How would the config need to look then?

Reply:
Mixed-up with the options:

```js
ClassicEditor
	.create( document.querySelector( '#snippet-custom-highlight-options' ), {
		toolbar: {
			items: [
				'headings', '|', 'highlightDropdown', 'bulletedList', 'numberedList', 'undo', 'redo'
			],
			viewportTopOffset: 60
		},
		highlight: {
			options: [
				{ model: 'greenMarker', class: 'marker-green', ... },
				{ model: 'bluePen', class: 'pen-blue', ... },
				'|',
				{ model: 'fooPen', class: 'pen-foo', ... }
				'|',
				'removeHighlight'
			]
		}
	} )
	.then( editor => {
		window.editor = editor;
	} )
	.catch( err => {
		console.error( err.stack );
	} );
```

Or as a separate property, which makes more sense to me because it doesn't mix up the content logic and the UI. 

In such situation, the order in `options` doesn't matter. We also provide a default `toolbar` layout, when no specified (and the default is `toolbar: [ all options, '|', 'removeHighlight' ]`)

```js
ClassicEditor
	.create( document.querySelector( '#snippet-custom-highlight-options' ), {
		toolbar: {
			items: [
				'headings', '|', 'highlightDropdown', 'bulletedList', 'numberedList', 'undo', 'redo'
			],
			viewportTopOffset: 60
		},
		highlight: {
			options: [
				{ model: 'greenMarker', class: 'marker-green', ... },
				{ model: 'bluePen', class: 'pen-blue', ... },
				{ model: 'fooPen', class: 'pen-foo', ... }
			],
			toolbar: [ 
				'highlight:greenMarker', 
				'highlight:bluePen', 
				'|', 
				'highlight:fooPen', 
				'|', 
				'removeHighlight' 
			]
		}
	} )
	.then( editor => {
		window.editor = editor;
	} )
	.catch( err => {
		console.error( err.stack );
	} );
```

Reply:
@oleq AFAIR that was an idea to have UI configurable using button instances (toolbarDropdown) and it should have issue for that. It is also doable the same way for alignment.

Reply:
@oleq  & @Reinmar : https://github.com/ckeditor/ckeditor5-ui/issues/322 & https://github.com/ckeditor/ckeditor5-heading/issues/74#issuecomment-305224952.

The way how alignment & highlight features works (buttons + dropdown) was inspired by that issue & comment - so thus enables it pretty straightforward.

Reply:
ps> I'd for doing this as a follow up just to see this feature live ;)

Reply:
> ps> I'd for doing this as a follow up just to see this feature live ;)

👍

Reply:
But I don't think that https://github.com/ckeditor/ckeditor5-ui/issues/322 has anything to do with this. Here, in this specific feature, there's more than just wrapping buttons into a dropdown, am I right? The state of the button, the color of the pen, etc, has some custom bindings.